secure_message/repository/retriever.py

```diff
@@ -230,9 +238,12 @@ def retrieve_populated_message_object(message_id: str) -> SecureMessage:
             if result is None:
                 logger.info("Message ID not found", message_id=message_id)
                 raise NotFound(description=f"Message with msg_id '{message_id}' does not exist")
-        except SQLAlchemyError:
+        except SQLAlchemyError as e:
+            if e.__class__ == NoResultFound:
```

Comment:
Will this ever happen? I'd have thought this would be dealt with by `if result is None` above

Reply:
I thought so too but when I was testing it the database threw the NoResultFound instead of returning None when I provided it with a uuid that wasn't in the database